Automattic/sensei
# Add saving of course structure

### Changes proposed in this Pull Request

* Save the course structure from the REST API request `http://example.com/wp-json/sensei-internal/v1/course-structure/{course_id}`. 
  * That request takes the structure from the request body (JSON) under `structure`.
  * For example:
```
{
   "structure":[
      {
         "type":"module",
         "id":50,
         "title":"Introduction",
         "description":"",
         "lessons":[
            {
               "type":"lesson",
               "id":1271,
               "title":"A-2 test lesson"
            },
            {
               "type":"lesson",
               "id":1261,
               "title":"A-1 test lesson"
            }
         ]
      },
      {
         "type":"module",
         "id":51,
         "title":"Mastery",
         "description":"Be really good at it",
         "lessons":[
            {
               "type":"lesson",
               "id":1272,
               "title":"B1 test lesson"
            },
            {
               "type":"lesson",
               "id":1273,
               "title":"B2 test lesson"
            }
         ]
      },
      {
         "type":"lesson",
         "id":1262,
         "title":"BB test lesson"
      }
   ]
}
```
* Updates how we hook into module/term save to override things.

### Testing instructions

* Verify setting courses when editing a module in WP Admin still works.
* Send various data to the REST API and verify that it works.
